src/runtime/loader.cs

```diff
@@ -49,13 +50,18 @@ public unsafe static int Initialize(IntPtr data, int size)
             return 0;
         }
 
-        public static int Shutdown(IntPtr data, int size)
+        public unsafe static int Shutdown(IntPtr data, int size)
         {
             IntPtr gs = IntPtr.Zero;
             try
             {
-                gs = PyGILState_Ensure();
-                PythonEngine.Shutdown();
+                var command = Encoding.UTF8.GetString((byte*)data.ToPointer(), size);
+
+                if (command == "full_shutdown")
+                {
+                    gs = PyGILState_Ensure();
+                    PythonEngine.Shutdown();
+                }
```

Comment:
I don't understand why pass `full_shutdown`/something else instead of calling/not calling `Shutdown` in the first place.

Reply:
This makes it easier to debug the issues for the time being, as it's one less recompile to compare both behaviours after fixing something. This parameter is completely temporary and will be gone once shutdown is stable.